### PR TITLE
Patch README

### PR DESCRIPTION
**Pull request by armandmgt**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,186 +23,165 @@ extern "C" {
...
+stageManager.request().hookToMiddle("MyModule", myHandlerFunction);
+```
+(See the following sections for an explanation of `request()` and `hookToMiddle`)
 
 ## StageManager
 
-StageManager will manage many Stages in your Server.
-There is four Stages implemented:<br/>
+StageManager should handle the different stages of a request process.
+Functions registered to each stage should be called when the following happens:
```
Comment (Valkyrihane) [inline README.md:45]: You should still let everyone know there are 4 different Stages

File: README.md [hunk 2]
```diff
@@ -22,186 +23,165 @@ extern "C" {
...
 
-## Stage
+## Stages
 
-| Moment      | Description |
-| :----:       |    :----:     |
-| **first** | It's called before middle when the event occurs (Example : SSL module)     |
-| **middle** | It's called before last during the event (Example : PHP module)       |
-| **last**  | It's called in the last moment of the event (Example : Encryption SSL module) |
+A stage is composed of three moments which permit the organization of modules in the specified stage.
```
Comment (Valkyrihane) [inline README.md]: "which permit" :(

File: README.md [hunk 3]
```diff
@@ -22,186 +23,165 @@ extern "C" {
...
+They are required through the use of the three function:
```
Comment (Valkyrihane) [inline README.md]: "of the three function" :(

File: README.md [hunk 4]
```diff
@@ -22,186 +23,165 @@ extern "C" {
...
 
-In the `Context` structure, the field `request` will contain the original request of the client.<br/>
-The field `response` should be filled by the different modules.<br/>
-We give you the field `socketFd` if you want to read or write data depending on the modules. 
-
-#### Struct HTTPMessage
+#### std::variant
 
+Here is an example of how to use a `std::variant`:
+(a std::variant is like an `union`, but type-safe)
```
Comment (Valkyrihane) [inline README.md]: maybe put a link to cppreference std::variant ?

File: README.md [hunk 5]
```diff
@@ -22,186 +23,165 @@ extern "C" {
...
+// This is how you retrieve the Request's firstLine property using std::get and log its path member.
 std::cout << std::get<dems::headers::Request>(context.request.firstLine).path << std::endl;
 ```
-On the fist line we show you how to create a `dems::Context`<br/>
-On the last line we take the path from the variant Request by using `std::get<T>`, `T` being the type you want (in this case, it is either `Request` or `Response`).
 
-The `headers` field contains a definition of Interface `IHeaders`
+#### Headers
 
+As headers are often used in the processing of a request, we choosed to impose the implementation of a container for headers. We made it look as simple as possible.
```
Comment (Valkyrihane) [inline README.md]: We choosed -> We chose

**Review by Valkyrihane — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (armandmgt) [reply README.md:45]: The stages are described in the table below, I don't think people are stupid.. are they ?

File: README.md (hunk 4 above)
Comment (armandmgt) [reply README.md]:
```suggestion
(a [std::variant](https://en.cppreference.com/w/cpp/utility/variant) is like an `union`, but type-safe)
```

File: README.md (hunk 5 above)
Comment (armandmgt) [reply README.md]:
```suggestion
As headers are often used in the processing of a request, we chose to impose the implementation of a container for headers. We made it look as simple as possible.
```

File: README.md [hunk 6]
```diff
@@ -22,186 +24,165 @@ extern "C" {
...
-    <body>
-        <h1>Hello World</h1>
-    </body>
-</html>"
-```
+---
 
 That's all !
 If you have any questions contact:
 [anatole.juge@epitech.eu](mailto:anatole.juge@epitech.eu?subject=G%201%20kestion%20sur%20la%20pays)
```
Comment (zombre98) [inline README.md]: I thought you wanted to remove these lines by discord link and Issues

**Review by zombre98 — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (Valkyrihane) [reply README.md:45]: Too much info doesn't kill the info

**Pull request merged**